sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -494,44 +494,54 @@ object DateTimeUtils {
   }
 
   /**
-   * Returns the month value for the given date. The date is expressed in days
-   * since 1.1.1970. January is month 1.
+   * Split date (expressed in days since 1.1.1970) into four fields:
+   * year, month (Jan is Month 1), dayInMonth, daysToMonthEnd (0 if it's last day of month).
    */
-  def getMonth(date: Int): Int = {
+  def splitDate(date: Int): (Int, Int, Int, Int) = {
```

Comment:
Oops, I mean `java.time` actually, but, I've also just realized that is only in Java 8, not Java 7. So yes I think the question here is whether it's more correct to use `java.util.Calendar`. I imagine you can make something faster, but is it going to be correct in all cases?